calculator-app/src/Components/Calculator.jsx

```diff
@@ -3,103 +3,144 @@ import Result from './Result'
 import CalculaterButton from './CalculatorButton'
 import History from './History'
 import '../style/calculator.css'
-const elements = ['1','2','3','+','4','5','6','-','7','8','9',':','*','^', 'CE','=','history']
 
 export default class Calculator extends React.Component {
   constructor(){
     super()
-    this.state= {result:"", history: [], isShowHistory: false}
+    this.state= {tmpRes:"",isPressed: false,isCalculated: false, result:"", history: [], isShowHistory: false}
     this.onClick = this.onClick.bind(this)
   }
-  onClick = action => {
-    let expectedResult = {}
+  async componentDidUpdate(prevProps){
+    let tmpSaveState = Object.assign({}, this.state)
+    if (!this.state.isCalculated && this.state.isPressed) {
+      let result = await this.calculate()
+      console.log(result)
+      if (!Number.isNaN(result) && typeof(result) != 'string') {
+        console.log(result)
+        let newState = Object.assign( tmpSaveState, {tmpRes: result, isCalculated : false, isPressed:false})
+        this.setState(newState)
+      }
+    } else if (this.state.isPressed) {
+      let newState = Object.assign( tmpSaveState,{isCalculated : false, isPressed:false} ) 
```

Comment:
Why don't you use this.setState({...this.state, isCalculated : false, isPressed:false}) ?